src/main/java/org/cyclonedx/util/TimestampUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of CycloneDX Core (Java).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.cyclonedx.util;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+@SuppressWarnings("unused")
+public final class TimestampUtils {
+  private static final SimpleDateFormat DEFAULT_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX");
+  private static final SimpleDateFormat ALT_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX");
```

Comment:
I had to add this extra pattern to avoid deserialization errors. One example can be found [here](https://github.com/shaikhu/cyclonedx-core-java/blob/master/src/test/resources/1.4/valid-vulnerability-1.4.json#L63..L65). 

I don't know if this alt date format was intended or not, but rather then editing all the input files, I just added this pattern as a fallback if the default format failed.

Reply:
I don't think we need to worry about dates here, I get your point, but maybe would make more sense to update the test files if they are different from the other, just to reduce complexity of this change

Reply:
> I don't think we need to worry about dates here, I get your point, but maybe would make more sense to update the test files if they are different from the other, just to reduce complexity of this change

Updated. fa67e9d1